pkg/plugins/runtime/gateway/route/util.go

```diff
@@ -23,18 +21,19 @@ func InferServiceProtocol(endpoints []core_xds.Endpoint, routeProtocol core_mesh
 			return core_mesh.ProtocolHTTP
 		}
 	}
```

Comment:
why this new if?

Reply:
Had a case when Gateway was pointing to a service that wasn't in the map and that caused an empty string